.github/workflows/build.yml

```diff
@@ -404,6 +402,9 @@ jobs:
 
   release:
     runs-on: ubuntu-24.04
+    permissions:
+      # required to create GitHub release
+      contents: write
```

Comment:
Same question here

Reply:
This overrides the global perm `contents: read` from the workflow within this job.

Reply:
Right, but I wondered if it _extended_ the top-level permissions or _overrides_ them. I know we added the `contents: read` because it was needed to run these in private forks (where `contents: read` is not available by default)

Reply:
I'm pretty sure it extends, I will check on runner app repo.

Reply:
You right it doesn't :man_facepalming:: https://docs.github.com/en/actions/writing-workflows/workflow-syntax-for-github-actions#permissions

> If you specify the access for any of these permissions, all of those that are not specified are set to `none`.